### DISCUSSION TITLE
How to use dropout

### DISCUSSION THREAD

**Discussion by MichaelMMeskhi** [google/flax#2655] [Q&A] [answered]
Based on the flax getting started page implementation, how would one go about adding a dropout layer? 

```python
class BaseNet(nn.Module):
    @nn.compact
    def __call__(self, x):
        x = nn.Dense(features=784)(x)
        x = nn.relu(x)
        x = nn.Dense(features=4096)(x)
        x = nn.relu(x)
        x = nn.Dense(features=10)(x)
        return x
```

And the training set is 

```python
@jax.jit
def train_step(state, batch):
  """Train for a single step."""
  def loss_fn(params):
    logits = BaseNet().apply({'params': params}, batch['image'])
    loss = cross_entropy_loss(logits=logits, labels=batch['label'])
    return loss, logits
  grad_fn = jax.grad(loss_fn, has_aux=True)
  grads, logits = grad_fn(state.params)
  state = state.apply_gradients(grads=grads)
  metrics = compute_metrics(logits=logits, labels=batch['label'])
  return state, metrics
```

Evaluation

```python
@jax.jit
def eval_step(params, batch):
  logits = BaseNet().apply({'params': params}, batch['image'])
  return compute_metrics(logits=logits, labels=batch['label'])
```

**Comment by 8bitmp3** [comment, marked as answer]
@MichaelMMeskhi the dropout tutorial is WIP:

https://github.com/google/flax/pull/2631/files

Let us know what you think.

**Reply by 8bitmp3** [reply]
Update:

The Dropout guide is live: https://flax.readthedocs.io/en/latest/guides/dropout.html
